tests/Unit/PrestaShopBundle/Api/StateProvider/HookStatusProviderTest.php

```diff
@@ -56,24 +58,27 @@ public function setUp(): void
         ;
     }
 
-    private function createResultBasedOnQuery(GetHookStatus $query): bool
+    private function createResultBasedOnQuery(GetHookStatus $query): HookStatus
     {
         if ($query->getHookId()->getValue() === 1) {
-            return false;
+            return new HookStatus($query->getHookId()->getValue(), false);
         }
 
         if ($query->getHookId()->getValue() === 2) {
-            return true;
+            return new HookStatus($query->getHookId()->getValue(), true);
         }
 
-        return true;
+        throw new RuntimeException(sprintf('Hook "%s" was not expected in query bus mock', $query->getHookId()->getValue()));
```

Comment:
Maybe in this case we could use a switch case default instead of multiples if and an implied default behavior

Reply:
I think it's enough as it meets all the use cases of the current test, it's even almost overkill

Reply:
More is better than less ^^